tools/internal/chartfetch/cmd.go

```diff
@@ -0,0 +1,167 @@
+package chartfetch
+
+import (
+	"errors"
+	"fmt"
+	"github.com/broadinstitute/terra-helmfile-images/tools/internal/flock"
+	"github.com/broadinstitute/terra-helmfile-images/tools/internal/shell"
+	"github.com/rs/zerolog"
+	"github.com/rs/zerolog/log"
+	"github.com/spf13/cobra"
+	"os"
+	"path"
+	"time"
+)
+
+const chartFetchDesc = `
+Fetch a chart from a Helm repo and unpack it into a user-supplied directory.
+
+This command is a wrapper around the "helm fetch" command. In fact, it just runs
+  helm fetch <chart> --version <version> -d /download/path/ --untar
+but with added safety (using flock) for multiple concurrent invocations.
+
+Examples:
+
+# Download version 1.2.3 of the agora chart to /tmp/charts/agora-v1.2.3
+chartfetch terra-helm/agora -v 1.2.3 -d /tmp/charts/agora-v1.2.3
+`
+
+// lockRetryInterval is the file lock retry interval
+const lockRetryInterval = 100 * time.Millisecond
+
+// lockTimeout is how long to wait to obtain a lock before giving up
+const lockTimeout = 10 * time.Second
+
+// Options for a chartfetch command
+type Options struct {
+	Version     string // Version of the chart that should be pulled (also passed directly to `helm fetch` )
+	DownloadDir string // DownloadDir Path where chart should be downloaded; if it exists, the chart will not be downloaded again
```

Comment:
It's `helmfile.yaml`'s responsibility to include the chart version in the download path. (Which it will, hope that was clear in the example).

The planned implementation will create a private temporary unpack directory for every `render` invocation, so even if there was some cache pollution/corruption it would go away after the render finished.